### DISCUSSION TITLE
'No such file or directory' Error when running scheduled github action

### DISCUSSION THREAD

**Discussion by mlampros** [community/community#27093] [Actions]
<p>Dear github actions support community,</p>
<p>I have a cron-job that runs once per week and executes an R script by taking as input a number of variables. If I push to the Github repository or click on the ‘Run’ button of the Github action then it runs successfully. However, it gives the error message ‘No such file or directory’ when it runs on the scheduled time, i.e. once per week. You can see the previous workflows in the following weblink:</p>
<p><a href="https://github.com/mlampros/fitbitVizBlog/actions" rel="noopener nofollow ugc">https://github.com/mlampros/fitbitVizBlog/actions</a></p>
<p>I tried various thinks to make it work, however none has worked. In the same .yml file there are multiple calls to the ‘Rscript’ command and although the majority work the specific ‘Rscript’ that takes the environment variables doesn’t.</p>
<p>I attach in this message also the error that I receive in form of a screenshot,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/1/e143453a281c6165732cd2fb0041f72b9208c2f1.png" data-download-href="/uploads/short-url/w8LyxkAvm3r2jZ7jQBJ14n8GN5D.png?dl=1" title="error_message" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/3055514/181101651-b7b5f5b9-0940-450f-bcf3-809dd7fcf786.png" alt="error_message" data-base62-sha1="w8LyxkAvm3r2jZ7jQBJ14n8GN5D" width="641" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/1/e143453a281c6165732cd2fb0041f72b9208c2f1_2_641x500.png, https://aws1.discourse-cdn.com/github/original/3X/e/1/e143453a281c6165732cd2fb0041f72b9208c2f1.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/e/1/e143453a281c6165732cd2fb0041f72b9208c2f1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/1/e143453a281c6165732cd2fb0041f72b9208c2f1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">error_message</span><span class="informations">720×561 55.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The ‘R_templates’ folder that includes the ‘data_template.R’ file is in the working directory of the github action step because I’ve specified:</p>
<pre><code class="lang-auto">working-directory: ${{ github.workspace }}
</code></pre>
<p>and I also checked this by using,</p>
<pre><code class="lang-auto">pwd
ls -a
</code></pre>
<p>Hopefully it’s something obvious and simple that I’m not aware of. I also took a look to previous similar issues with the same error message (‘No such file or directory’ ) where the issues were related to the upper and lower case sensitivity, however I’ve used the exact names of folders and files in my .yml file.</p>
<p>Thank you in advance and for your time.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>This part of the workflow is your problem:</p>
<pre><code class="lang-yaml">    steps:
      - uses: actions/checkout@v2
        if: github.event_name == 'pull_request'
        with:
          fetch-depth: 0
          ref: ${{ github.event.pull_request.head.ref }}
      - uses: actions/checkout@v2
        if: github.event_name == 'push'
        with:
          fetch-depth: 0
</code></pre>
<p>The <code>if</code> conditions on the two <code>actions/checkout</code> effectively mean that checkout will only happen for push or pull_request events, not on the scheduled run. So your repository data isn’t available. One simple fix would be to change the second condition to <code>github.event_name != 'pull_request'</code>.</p>

**Comment by mlampros** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> thank you very match, your answer was the solution to the mentioned issue.</p>
